assets/src/edit-story/components/library/libraryProvider.js

```diff
@@ -74,6 +74,25 @@ function LibraryProvider({ children }) {
     }
   }, [highlightedTab]);
 
+  const mediaTabRef = useRef(null);
+  const media3pTabRef = useRef(null);
+  const textTabRef = useRef(null);
+  const shapesTabRef = useRef(null);
+  const elementsTabRef = useRef(null);
+  const pageTemplatesTabRef = useRef(null);
+
+  const tabRefs = useMemo(
+    () => ({
+      [MEDIA.id]: mediaTabRef,
+      [MEDIA3P.id]: media3pTabRef,
+      [TEXT.id]: textTabRef,
+      [SHAPES.id]: shapesTabRef,
+      [ELEMS.id]: elementsTabRef,
+      [PAGE_TEMPLATES.id]: pageTemplatesTabRef,
+    }),
+    []
+  );
```

Comment:
This feels really gross to me but I don't know of a way to create individual refs dynamically. It feels like we have to do this because `useFocusOut` expects the ref to look like `{ current: <element> }`

Reply:
Yeah, personally I'm ok with it. There's a difference in expectations of how refs get passed. This seems like the best place to set up something like this and again, it's not making it harder to read through. I don't think it's gross! 😝

Reply:
This is the same reason I created a new hook in highlights called `useFocusHighlight` which takes a ref so we don't have to track them in the hook itself.

Reply:
I think I'm confused. I'm setting these refs up here so that I can pass them to `useFocusHighlight` 😕. is that wrong?

Reply:
Sorry, I was confused by the intention of the changes here and so ignore my above comment, apologies.

I do think there is an issue with focusing the tab as a result of the tab changing. We now have to pass the refs down to the TabView to do that, coupling the tab controllers with the tab component, which is used generally for the library and inspector.

We actually want to focus the tabs as a result of a quick action, rather than as a result of the tab changing. Since quick actions are the hook that is concerned with focusing the tabs, we should pass the refs to the hook or a wrapper for the hook which deals with focusing refs on certain quick actions, rather than changing the functionality of the tab view.